forge-gui/res/cardsfolder/upcoming/mapping_the_maze.txt

```diff
@@ -0,0 +1,9 @@
+Name:Mapping the Maze
+ManaCost:1 U
+Types:Instant
+A:SP$ ChooseCard | ChoiceZone$ Hand,Graveyard | Choices$ Instant.YouOwn,Sorcery.YouOwn | ChoiceTitle$ Choose an instant or sorcery card in your hand or graveyard | SubAbility$ DBAnimate | SpellDescription$ Choose an instant or sorcery card in your hand or graveyard. 
+SVar:DBAnimate:DB$ Animate | Triggers$ SpellCastTrig | Defined$ ChosenCard | Incorporate$ U | Duration$ Perpetual | SubAbility$ DBReturn | SpellDescription$ That card perpetually incorporates {U} and gains "When you cast this spell, draw a card."
+SVar:DBReturn:DB$ ChangeZone | Origin$ Graveyard | Destination$ Hand | ChangeTypeDesc$ a multicolored instant or sorcery card from your graveyard | ChangeNum$ 1 | Hidden$ True | ChangeType$ Instant.YouCtrl+MultiColor,Sorcery.YouCtrl+MultiColor | SpellDescription$ Then return a multicolored instant or sorcery card from your graveyard to your hand.
+SVar:SpellCastTrig:Mode$ SpellCast | ValidCard$ Card.Self | Execute$ TrigDraw | TriggerDescription$ When you cast this spell, draw a card.
+SVar:TrigDraw:DB$ Draw
+Oracle:Choose an instant or sorcery card in your hand or graveyard. That card perpetually incorporates {U} and gains "When you cast this spell, draw a card." Then return a multicolored instant or sorcery card from your graveyard to your hand.
```

Comment:
DeckNeeds Instant Sorcery, right?
DeckHas Graveyard
I don't know if there's a good AI flag that this should be in a multicolor deck
If not we may RemoveDeck:Random ?